src/controllers/dynakube/dynakube_controller_test.go

```diff
@@ -187,15 +93,48 @@ func TestReconcileActiveGate_Reconcile(t *testing.T) {
 
 		kubeMonCapability := capability.NewKubeMonCapability(instance)
 		name := capability.CalculateStatefulSetName(kubeMonCapability, instance.Name)
-		err = fakeClient.Get(context.TODO(), client.ObjectKey{Name: name, Namespace: testNamespace}, &statefulSet)
+		err = r.client.Get(context.TODO(), client.ObjectKey{Name: name, Namespace: testNamespace}, &statefulSet)
 
 		assert.NoError(t, err)
 		assert.NotNil(t, statefulSet)
 	})
 }
 
+func TestReconcileOnlyOneTokenProvided_Reconcile(t *testing.T) {
+	t.Run(`Reconcile validates apiToken correctly if apiToken with "InstallerDownload"-scope is provided`, func(t *testing.T) {
+		mockClient := createDTMockClient(dtclient.TokenScopes{},
+			dtclient.TokenScopes{dtclient.TokenScopeDataExport, dtclient.TokenScopeInstallerDownload})
+
+		instance := &dynatracev1beta1.DynaKube{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      testName,
+				Namespace: testNamespace,
+			},
+			Spec: dynatracev1beta1.DynaKubeSpec{}}
+		r := createFakeClientAndReconcile(mockClient, instance, "", testAPIToken)
+
+		result, err := r.Reconcile(context.TODO(), reconcile.Request{
+			NamespacedName: types.NamespacedName{Namespace: testNamespace, Name: testName},
+		})
+
+		assert.NoError(t, err)
+		assert.NotNil(t, result)
+
+		var secret corev1.Secret
+
+		err = r.client.Get(context.TODO(), client.ObjectKey{Name: testName, Namespace: testNamespace}, &secret)
+
+		assert.NoError(t, err)
+		assert.NotNil(t, secret)
+		assert.Equal(t, string(secret.Data[dtclient.DynatracePaasToken]), testAPIToken)
+		assert.Equal(t, string(secret.Data[dtclient.DynatraceApiToken]), testAPIToken)
+	})
+}
+
 func TestReconcile_RemoveRoutingIfDisabled(t *testing.T) {
-	mockClient := &dtclient.MockDynatraceClient{}
+	// replacing is possible
```

Comment:
replacing what?

Reply:
Was just a hint for me during development, will be removed